packages/compiler/src/expression_parser/parser.ts

```diff
@@ -1005,25 +1020,32 @@ export class _ParseAST {
   }
 
-  // Error recovery should skip tokens until it encounters a recovery point. skip() treats
-  // the end of input and a ';' as unconditionally a recovery point. It also treats ')',
-  // '}' and ']' as conditional recovery points if one of calling productions is expecting
-  // one of these symbols. This allows skip() to recover from errors such as '(a.) + 1' allowing
-  // more of the AST to be retained (it doesn't skip any tokens as the ')' is retained because
-  // of the '(' begins an '(' <expr> ')' production). The recovery points of grouping symbols
-  // must be conditional as they must be skipped if none of the calling productions are not
-  // expecting the closing token else we will never make progress in the case of an
-  // extraneous group closing symbol (such as a stray ')'). This is not the case for ';' because
-  // parseChain() is always the root production and it expects a ';'.
-
-  // If a production expects one of these token it increments the corresponding nesting count,
-  // and then decrements it just prior to checking if the token is in the input.
+  /**
+   * Error recovery should skip tokens until it encounters a recovery point. skip() treats
+   * the end of input and a ';' as unconditionally a recovery point. It also treats ')',
+   * '}' and ']' as conditional recovery points if one of calling productions is expecting
+   * one of these symbols. This allows skip() to recover from errors such as '(a.) + 1' allowing
+   * more of the AST to be retained (it doesn't skip any tokens as the ')' is retained because
+   * of the '(' begins an '(' <expr> ')' production). The recovery points of grouping symbols
+   * must be conditional as they must be skipped if none of the calling productions are not
+   * expecting the closing token else we will never make progress in the case of an
+   * extraneous group closing symbol (such as a stray ')'). This is not the case for ';' because
+   * parseChain() is always the root production and it expects a ';'.
+   *
+   * Furthermore, the presence of a stateful context can add more recovery points.
+   *   - in a `writeContext`, we are able to recover after seeing the `=` operator, which signals
+   *     the presence of an independent rvalue expression following the `=` operator.
+   *
+   * If a production expects one of these token it increments the corresponding nesting count,
+   * and then decrements it just prior to checking if the token is in the input.
+   */
   private skip() {
     let n = this.next;
     while (this.index < this.tokens.length && !n.isCharacter(chars.$SEMICOLON) &&
            (this.rparensExpected <= 0 || !n.isCharacter(chars.$RPAREN)) &&
            (this.rbracesExpected <= 0 || !n.isCharacter(chars.$RBRACE)) &&
-           (this.rbracketsExpected <= 0 || !n.isCharacter(chars.$RBRACKET))) {
+           (this.rbracketsExpected <= 0 || !n.isCharacter(chars.$RBRACKET)) &&
+           !(this.writeContext && n.isOperator('='))) {
```

Comment:
```suggestion
           !(this.writeContext && n.isOperator('='))) {
```

The direction of the comparison here is a little jarring when combined with the conditions above. It might read better if it were `(!this.writeContext || !n.isOperator('='))`. I don't feel super strongly about this though.

Reply:
I find the inverted version `!this.writeContext || !n.isOperator('=')` more readable, too, and it fits with the style of the other conditions that are and'ed together here.

Reply:
Done